core/dbt/events/types.py

```diff
@@ -256,6 +256,22 @@ def message(self) -> str:
         return line_wrap_message(warning_tag(f"Deprecated functionality\n\n{description}"))
 
 
+class PackageRedirectDeprecationSummary(WarnLevel):
+    def code(self) -> str:
+        return "D021"
+
+    def message(self) -> str:
+        description = (
+            f"Found {self.occurrences} packages that have been deprecated in favor of new packages. Please "
```

Comment:
Can occurrences be 1? Do you need to deal with plurality here?

Reply:
They can, I'll get this updated 🙂